### PR TITLE
created survey API's

### PR DESCRIPTION
**Pull request by deepakr123**

### CODE REVIEW

File: helper/chart_data.js [hunk 1]
```diff
@@ -8,23 +8,33 @@ const default_no_of_assessment_submissions_threshold = 3;
...
+            type = filesHelper.survey;
+            obj = {
+                solutionName: data[0].event.solutionName,
+                response: []
+            }
+        }
+        else {
+            type = filesHelper.observation;
+            obj = {
+                entityName: data[0].event.schoolName,
```
Comment (aks30) [inline helper/chart_data.js]: @deepakr123 - Please confirm is this is being used. If not let's not send it.

File: helper/chart_data.js [hunk 2]
```diff
@@ -118,15 +128,15 @@ exports.instanceReportChart = async function (data,reportType) {
 
          //loop the keys of matrix array
          await Promise.all(matrixRes.map(async ele => {
-            let matrixResponse = await matrixResponseObjectCreateFunc(matrixResult[ele])
+            let matrixResponse = await matrixResponseObjectCreateFunc(matrixResult[ele], type)
```
Comment (aks30) [inline helper/chart_data.js]: @deepakr123 - Plan for a discussion on this with Shakthi on Monday.

File: helper/chart_data.js [hunk 3]
```diff
@@ -181,49 +191,59 @@ async function instanceMultiselectFunc(data) {
...
+    let multiSelectArray = [];
+    let textArray = [];
+    let radioArray = [];
+    let sliderArray = [];
+    let numberArray = [];
+    let dateArray = [];
+    let noOfSubmissions = [];
+    let matrixArray = [];
     let actualData = data;
+    let type;
```
Comment (aks30) [inline helper/chart_data.js]: @deepakr123 - Please rename it to solutionType

File: helper/chart_data.js [hunk 4]
```diff
@@ -181,49 +191,59 @@ async function instanceMultiselectFunc(data) {
...
-                observationName: data[0].event.observationName,
-                entityType: data[0].event.entityType,
-                entityId: entityId,
-                entityName:  data[0].event[entityName + "Name"],
-                districtName: data[0].event.districtName,
-                response: []
+            type = filesHelper.observation;
+            if (data[0].event.solutionId) {
+
+                obj = {
```
Comment (aks30) [inline helper/chart_data.js]: @deepakr123 - Please change to meaningful variable name

File: helper/chart_data.js [hunk 5]
```diff
@@ -181,49 +191,59 @@ async function instanceMultiselectFunc(data) {
...
+                    solutionId: data[0].event.solutionId,
+                    solutionName: data[0].event.solutionName,
+                    entityType: data[0].event.entityType,
```
Comment (aks30) [inline helper/chart_data.js]: @deepakr123 - make a common object and then add keys based on if condition

File: migrations/20200913230617-insert_survey_solutions_report_query.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    if (!row.length) {
+
+    let id = Uuid.random();
+
+    let query = 'INSERT INTO ' + config.cassandra.keyspace + '.' + config.cassandra.table +' (id, qid, query) VALUES (?, ?, ?)';
+   
+    let queries = 
+      [{
+        query: query,
+        params: [id.toString(), 'survey_solutions_report_query','{"queryType":"groupBy","dataSource":"sl_survey","granularity":"all","dimensions":["surveySubmissionId","solutionId","solutionName","questionName","questionAnswer","remarks","surveyId","questionResponseType","questionResponseLabel","questionId","questionExternalId","instanceId","instanceParentQuestion","instanceParentResponsetype","instanceParentId","questionSequenceByEcm","instanceParentExternalId","instanceParentEcmSequence","completedDate"],"filter":{"type":"and","fields":[{"type":"selector","dimension":"solutionId","value":""}]},"aggregations":[],"postAggregations":[],"limitSpec":{"type":"default","limit":10000,"columns":[{"dimension":"questionExternalId","direction":"ascending"}]},"intervals":["1901-01-01T00:00:00+00:00/2101-01-01T00:00:00+00:00"]}']
```
Comment (aks30) [inline migrations/20200913230617-insert_survey_solutions_report_query.js]: @deepakr123 - Can you share me the response for 100+ submissions made for a survey ? Need to load only aggregated report from duid and first 10 for text questions.

File: controllers/v1/surveys.js [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+
+                        let message;
+                        let getSubmissionStatusResponse = await assessmentService.getSurveySubmissionStatusById
+                        (
+                            req.query.submissionId
+                        )
+
+                        if (getSubmissionStatusResponse.result && 
+                            getSubmissionStatusResponse.result.status == filesHelper.submission_status_completed) {
+                            message = "Report is being created, please check after a while"
```
Comment (aks30) [inline controllers/v1/surveys.js]: @deepakr123 - Change the text to - "Your report generation is in progress. Please check back after a few minutes."

Comment (deepakr123) [reply controllers/v1/surveys.js]: resolved

File: helper/chart_data.js (hunk 4 above)
Comment (deepakr123) [reply helper/chart_data.js]: resolved

File: helper/chart_data.js (hunk 5 above)
Comment (deepakr123) [reply helper/chart_data.js]: resolved

File: helper/chart_data.js (hunk 3 above)
Comment (deepakr123) [reply helper/chart_data.js]: resolved

File: helper/chart_data.js (hunk 1 above)
Comment (deepakr123) [reply helper/chart_data.js]: resolved

File: helper/chart_data.js (hunk 2 above)
Comment (deepakr123) [reply helper/chart_data.js]: resolved

File: controllers/v1/surveys.js [hunk 8]
```diff
@@ -0,0 +1,614 @@
...
+        }]
+   *  }
+   * @apiUse errorBody
+   */
+
+   exports.listAllEvidences = async function (req, res) {
+
+    return new Promise(async function (resolve, reject) {
+
+        if (!req.body.submissionId && !req.body.questionId) {
```
Comment (aks30) [inline controllers/v1/surveys.js]: @deepakr123 - Please fix this condition

**Pull request merged**